### PR TITLE
Enable SecondaryCache::CreateFromString to create sec cache based on the uri for CompressedSecondaryCache

### PR DESCRIPTION
**Pull request by gitbw95**
Summary:
Update SecondaryCache::CreateFromString and enable it to create sec cache based on the uri for CompressedSecondaryCache.

Test Plan:
Add unit tests.

### CODE REVIEW

File: cache/cache.cc [hunk 1]
```diff
@@ -34,13 +34,65 @@ static std::unordered_map<std::string, OptionTypeInfo>
...
     const ConfigOptions& config_options, const std::string& value,
     std::shared_ptr<SecondaryCache>* result) {
-  return LoadSharedObject<SecondaryCache>(config_options, value, nullptr,
-                                          result);
+  if (value.find("compressed_secondary_cache://") == 0) {
+    std::string args = value;
+    args.erase(0, std::strlen("compressed_secondary_cache://"));
+    Status status;
+    std::shared_ptr<SecondaryCache> sec_cache;
+
```
[the remark this answers is not in the document]
Comment (gitbw95) [reply cache/cache.cc:68]: This PR uses the similar way as Cache::CreateFromString to unblock PR #10042 ("Add the compressed secondary cache option into stress test"). In one comment of #10042, you mentioned "I plan to address those in a future PR (unassociated with this PR).".

File: cache/cache.cc [hunk 2]
```diff
@@ -34,13 +34,65 @@ static std::unordered_map<std::string, OptionTypeInfo>
...
+static std::unordered_map<std::string, OptionTypeInfo>
+    comp_sec_cache_options_type_info = {
+        {"capacity",
+         {offsetof(struct CompressedSecondaryCacheOptions, capacity),
+          OptionType::kSizeT, OptionVerificationType::kNormal,
+          OptionTypeFlags::kMutable}},
+        {"num_shard_bits",
+         {offsetof(struct CompressedSecondaryCacheOptions, num_shard_bits),
+          OptionType::kInt, OptionVerificationType::kNormal,
+          OptionTypeFlags::kMutable}},
```
[the remark this answers is not in the document]
Comment (gitbw95) [reply cache/cache.cc:47]: This CompressedSecondaryCache is a concrete implementation of SecondaryCache on top of LRUCache with its own CompressedSecondaryCacheOptions.

[the remark this answers is not in the document]
Comment (gitbw95) [reply cache/cache.cc:47]: @mrambacher Thanks for the recommendation! The "cleaner way" can be one of the options to optimize current CompressedSecondaryCache in future, but it doesn't have the highest priority now. With the current  CompressedSecondaryCache, we would like to firstly add stress tests, analyze and minimize memory fragmentations, analyze and prevent the re-compression of cold blocks, etc.

**Review by anand1976 — APPROVED** [review]
LGTM. Creating the compressed secondary cache through the object registry will be tackled in a future PR.

**Pull request closed without merging**